File: WebsiteHoTroDichVuBanDoc_Backend/WebsiteHoTroDichVuBanDoc_Backend/app/models/dat_phong.py
from typing import Optional
from datetime import datetime
from app.models.db_base import DBModel

class DatPhongBase(DBModel):
    maPhong: int
    nguoiToChuc: str
    soDienThoai: str
    thoiGianBatDau: datetime
    thoiGianKetThuc: datetime
    mucDichSuDung: Optional[str] = None
    soNguoiThamDuDuKien: int = 0
    trangThai: str = 'kichHoat'
    maNhanVien: Optional[int] = None # Nhân viên duyệt

class DatPhongCreate(DatPhongBase):
    pass

class DatPhongUpdate(DBModel):
    trangThai: Optional[str] = None
    maNhanVien: Optional[int] = None

class DatPhong(DatPhongBase):
    maDatPhong: int
    ngayKhoiTao: Optional[datetime] = None

class DatPhongDuyet(DBModel):
    """Model input cho hành động duyệt phòng."""
    maNhanVien: int

File: WebsiteHoTroDichVuBanDoc/Backend/src/main/java/com/library/website/models/DatPhong.java
package com.library.website.models;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "datphong", schema = "public")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class DatPhong {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "madatphong")
    private Long maDatPhong;

    @ManyToOne
    @JoinColumn(name = "maphong")
    private Phong phong;

    @Column(name = "nguoitochuc", nullable = false)
    private String nguoiToChuc;

    @Column(name = "sodienthoai", nullable = false)
    private String soDienThoai;

    @Column(name = "thoigianbatdau", nullable = false)
    private LocalDateTime thoiGianBatDau;

    @Column(name = "thoigianketthuc", nullable = false)
    private LocalDateTime thoiGianKetThuc;

    @Column(name = "mucdichsudung")
    private String mucDichSuDung;

    @Column(name = "songuoithamdudukien")
    private Integer soNguoiThamDuDuKien = 0;

    @Column(name = "trangthai")
    private String trangThai = "kichHoat"; // 'kichHoat', 'daHuy', 'hoanThanh'

    @Column(name = "manhanvien")
    private Long maNhanVien;

    @Column(name = "ngaykhoitao")
    private LocalDateTime ngayKhoiTao = LocalDateTime.now();

    @Column(name = "updated_at")
    private LocalDateTime updatedAt = LocalDateTime.now();

    @PreUpdate
    public void preUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}
